apitests/spec/devices_spec.ts

```diff
@@ -24,11 +54,108 @@ describe('devices api', () => {
         });
 
         it('should return system properties', (done) => {
-            request.get('', (err, resp, result) => {
+            request.get('', (err, resp, result:Devices) => {
                 expect(result.data[0].SystemProperties).toBeTruthy();
                 expect(result.data[0].SystemProperties.ICCID).toBeDefined();
                 done();
             });
         });
+
+        it('should always have required properties', (done) => {
+            request.get('', (err, resp, result:Devices) => {
+                expect(result.data[0].DeviceProperties).toBeTruthy();
+                expect(result.data[0].SystemProperties).toBeTruthy();
+                expect(result.data[0].Commands).toBeTruthy();
+                expect(result.data[0].CommandHistory).toBeTruthy();
+                expect(result.data[0].IsSimulatedDevice).toBeDefined();
+                expect(result.data[0].id).toBeTruthy();
+                expect(result.data[0]._rid).toBeTruthy();
+                expect(result.data[0]._self).toBeTruthy();
+                expect(result.data[0]._etag).toBeTruthy();
+                expect(result.data[0]._ts).toBeTruthy();
+                expect(result.data[0]._attachments).toBeTruthy();
+                done();
+            });
+        })
+
+        it('should have required attributes for enabled devices', (done) => {
+            request.get('', (err, resp, result:Devices) => {
+                let device:DeviceInfo = findEnabledDevice(result);
```

Comment:
expect device to be a valid object to guard against case where no enabled devices are found.